src/Features/LanguageServer/Protocol/Handler/Completion/CompletionResolveHandler.cs

```diff
@@ -185,5 +188,40 @@ private static bool MatchesLSPCompletionItem(LSP.CompletionItem lspCompletionIte
 
             return cacheEntry;
         }
+
+        // Certain language servers such as Razor may want TextEdits formatted using their own options instead of C#/VB options.
+        private static async Task<OptionSet> GetCompletionFormattingOptionsAsync(Document document, CancellationToken cancellationToken)
```

Comment:
A document options provider for razor documents seems like a sound approach to me, but if we're going that route, I think we should bake it into where the options are used (e.g. the formatter).  
That way any LSP feature that cares about razor options just does the right thing automatically.  With the current implementation, we have to repeat this code on every LSP feature and implementers have to be aware of the razor case otherwise it is by default broken.